src/electionguard/election_object_base.py

```diff
@@ -1,8 +1,32 @@
+"""Base objects to derive other election objects."""
+
 from dataclasses import dataclass
+from typing import List, TypeVar
 
 
 @dataclass
 class ElectionObjectBase:
     """A base object to derive other election objects identifiable by object_id."""
 
     object_id: str
+
+
+@dataclass
+class OrderedObjectBase(ElectionObjectBase):
```

Comment:
This new base helps identify which objects can be ordered. It also ensures that the first two elements in the constructor are always `object_id` then `sequence_order` to avoid confusion. This did cause a ripple effect through a couple other classes.